pandas/_libs/interval.pyx

```diff
@@ -299,7 +299,6 @@ cdef class Interval(IntervalMixin):
     True
     """
     _typ = "interval"
-    __array_priority__ = 1000
```

Comment:
i couldve sworn there was a reason this was set, surprised removing it doesnt break anything

Reply:
It does serve the purpose of making np.timedelta64 + pd.Interval(pd.Timestamp, pd.Timestamp) work, but apparently that was never actually tested. I figure this is more of an edge case compared to the bug / change in numpy behavior mentioned in the issue.

1.1.1
```python
In [1]: import numpy as np
   ...: import pandas as pd
   ...:
   ...: delta = np.timedelta64(0)
   ...: interval = pd.Interval(pd.Timestamp("2020-01-01"), pd.Timestamp("2020-02-01"))
   ...: delta + interval
   ...:
Out[1]: Interval('2020-01-01', '2020-02-01', closed='right')
```

This branch
```python
In [1]: import numpy as np
   ...: import pandas as pd
   ...:
   ...: delta = np.timedelta64(0)
   ...: interval = pd.Interval(pd.Timestamp("2020-01-01"), pd.Timestamp("2020-02-01"))
   ...: delta + interval
   ...:
---------------------------------------------------------------------------
UFuncTypeError                            Traceback (most recent call last)
<ipython-input-1-2c3b3a0b4fb8> in <module>
      4 delta = np.timedelta64(0)
      5 interval = pd.Interval(pd.Timestamp("2020-01-01"), pd.Timestamp("2020-02-01"))
----> 6 delta + interval

UFuncTypeError: ufunc 'add' cannot use operands with types dtype('<m8') and dtype('O')
```

Oddly though the other direction still works

```python
In [2]: interval + delta
Out[2]: Interval('2020-01-01', '2020-02-01', closed='right')
```

I could open a separate issue for this in case someone can think of another solution.

Reply:
I added this as an xfailing test for now

Reply:
I agree that the replace and working array broadcasting seems more important than the addition of intervals. So +1 on this PR. 

@dsaxton can you open an issue about supporting the now removed behaviour?

Reply:
@jorisvandenbossche Created https://github.com/pandas-dev/pandas/issues/35961

Reply:
Closed the issue because Brock's suggestion fixes this without causing another regression